pservice/lib/libpdo_enclave/secret_enclave.cpp

```diff
@@ -676,14 +677,24 @@ pdo_err_t VerifyEnclaveInfo(const std::string& enclaveInfo,
         memcmp(computedReportData.d, expectedReportData.d, SGX_REPORT_DATA_SIZE)  != 0,
         "Invalid Report data: computedReportData does not match expectedReportData");
 
-    //Note that we do not currently verify whether the enclave debug flag is
-    //turned on or not. In order to ensure that the enclave is run in a mode
-    //that supports enhanced-confidentiality and execution integrity, the debug
-    //flag (SGX_FLAGS_DEBUG / 0x0000000000000002ULL in the report's attribute)
-    //should be set to 0. For additional details on how we plan to support this
-    //check, please see
-    //https://github.com/hyperledger-labs/private-data-objects/issues/195.
-    //
+    // Verify 64-bit enclave
+    pdo::error::ThrowIf<pdo::error::ValueError>((attributes.flags & SGX_FLAGS_MODE64BIT) == 0,
+            "Invalid 64-bit flag: 0");
+
+    // Verify SGX debug flag: check mask and enforce if necessary
+    if(attribute_mask_flags & SGX_FLAGS_DEBUG) //if bit is set, enforce debug flag check
```

Comment:
Good question, and the answer is in the generate-mrenclave-header.

The mask comes from the metadata, generated at enclave build time, from which we extract the mrenclave provided to the pservice. It is necessary because the sgx-sign disables/enables debug through the mask (which says what flags must be enforced). If the debug flag in the mask is not set, then the debug flag is irrelevant (meaning that the eservice can start the enclave in both modes). The DisableDebug field in the enclave xml file allows to modify the mask accordingly (and if debug is disabled, the debug flag is forced to 0). With this PR the DisableDebug field depends on PDO_DEBUG_BUILD.

The quote and the attestation are instead runtime artifacts which will contain the actual debug flag use to start the enclave. The TP enforces that based on the policy, which this PR modifies by adding the debug flag. Also, in HW mode, the TP can be run the policy debug flag 0/1 based on PDO_DEBUG_BUILD.

Reply:
Oh, mea culpa, i missed that the test in the if body is conditional (and on the extracted flag). So i see now how (and that) it works.  But maybe could still be useful to take some of the text above and as a comment before the attribute_mask test?